gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -162,13 +170,20 @@ private boolean isBranchPresent(String branchFullName) {
     // the branch name (for `refs/heads/<branch_name>`)
     for (int numOfSegmentsToUse = 3; numOfSegmentsToUse < segments.size(); numOfSegmentsToUse++) {
       val testedPrefix = segments.take(numOfSegmentsToUse).mkString("/");
-      if (Try.of(() -> jgitRepoForMainGitDir.resolve(testedPrefix)).getOrNull() != null) {
-        return false;
-      }
+      try {
+        val objectID = jgitRepoForMainGitDir.resolve(testedPrefix);
+        if (objectID != null) {
+          return false;
+        }
+      } catch (IOException ignored) {}
```

Comment:
Hmmm the behavior here equivalent to what was previously? 🤔

Reply:
This was a tricky one, but from what I understood it goes like this:

When result of `jgitRepoForMainGitDir` is:
- **nonnull** return false.
- **null** or **exception** ignore it and move forward.

Then for the next one (line 171) result of `jgitRepoForMainGitDir` is:
- any result resturn result != null
- **exception** return false

Please correct me if I'm wrong

Reply:
Awww right. Indeed the logic is equivalent, no action needed